system-int-test/src/cmr/system_int_test/utils/service_util.clj

```diff
@@ -0,0 +1,120 @@
+  ; (util/assert-convert-kebab-case [:concept-id :revision-id
+  ;                                  :service-name :originator-id]
+  ;                                 data))
+  true)
+
+(defn grant-all-service-fixture
+  "A test fixture that grants all users the ability to create and modify
+  services."
+  [f]
+  (echo-util/grant-all-service (s/context))
+  (f))
+
+(def sample-service
+  {:Name "A name"
+   :Description "A UMM-S description"
+   :Format ""
+   :Type "Web Mapping Service"
+   :Version "1.1.1"})
+
+(defn make-service
+  "Makes a valid service based on the given input."
+  ([]
+   (make-service nil))
+  ([attrs]
+   (merge sample-service attrs))
+  ([index attrs]
+   (merge
+    sample-service
+    {:Name (str "Name" index)
+     :Version (str "V" index)
+     :Description (str "UMM-S description " index)}
+    attrs)))
+
+(defn create-service
+  "Creates a service."
+  ([token service]
+   (create-service token service nil))
+  ([token service options]
+   (let [options (merge {:raw? true :token token} options)]
+     (ingest-util/parse-map-response
+      (transmit-service/create-service (s/context) service options)))))
+
+(defn update-service
+  "Updates a service."
+  ([token service]
+   (update-service token (:service-id service) service nil))
+  ([token service-id service]
+   (update-service token service-id service nil))
+  ([token service-id service options]
+   (let [options (merge {:raw? true :token token} options)]
+     (ingest-util/parse-map-response
+      (transmit-service/update-service (s/context) service-id service options)))))
+
+(defn save-service
+  "A helper function for creating or updating services for search tests.
+
+   If the service does not have a :concept-id, it saves it. If the service has
+   a :concept-id, it updates the service. Returns the saved service along
+   with :concept-id, :revision-id, :errors, and :status."
+  [token service]
+  (let [service-to-save (select-keys service [:service-name :description :revision-date])
+        response (if-let [concept-id (:concept-id service)]
+                   (update-service token (:service-name service) service-to-save)
+                   (create-service token service-to-save))
+        service (-> service
+                    (update :service-name string/lower-case)
+                    (into (select-keys response [:status :errors :concept-id :revision-id])))]
+    (if (= (:revision-id service) 1)
+      ;; Get the originator id for the service
+      (assoc service :originator-id (tokens/get-user-id (s/context) token))
+      service)))
+
+(defn assert-service-saved
+  "Checks that a service was persisted correctly in metadata db. The service
+  should already have originator id set correctly. The user-id indicates which
+  user updated this revision."
+  [service user-id concept-id revision-id]
+  (let [concept (mdb/get-concept concept-id revision-id)]
+    (is (= {:concept-type :service
+            :native-id (:service-name service)
```

Comment:
This was copied from your work of `cmr.system-int-test.utils.variable-util/assert-variable-saved`; I will update that one too.